### DISCUSSION TITLE
Code from "third_party" folder missing from results

### DISCUSSION THREAD

**Discussion by jlisee** [community/community#11176] [Other Feature Feedback, Questions, & Ideas] [answered]
In our private repository on cs.github.com I can browse to files in our the `third_party` folder in the root of our repository but searches do not return any content there.  I can even visit a file and see a detected symbol then click "search for .... in this repo" and it finds it everywhere *except* the file I was just in.

Is there some global filtering control that is taking effect here that can be adjusted?

**Comment by colinwm** [comment, marked as answer]
Hey @jlisee, we currently don't index all code, see https://cs.github.com/about/faq#indexed-content. Code in the `third_party` directory is getting detected as vendored, so it isn't indexed.

We're working on removing that restriction, but for now it's not searchable. Sorry about that!

**Reply by traversaro** [reply]
@colinwm the link https://cs.github.com/about/faq#indexed-content is not working anymore, perhaps there is an updated reference for this restriction? Thanks!

**Reply by LiteBrite82** [reply]
I couldn't find the equivalent, I think I came close though: [About GitHub Code Search: Limitations](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations). Or maybe [Finding files on GitHub](https://docs.github.com/en/search-github/searching-on-github/finding-files-on-github)

I also found a blog with an in depth explanation: [The technology behind GitHub’s new code search](https://github.blog/2023-02-06-the-technology-behind-githubs-new-code-search/). It may be a bit TL;DR though...

If you really want to get in depth, check out ["Lessons from building GitHub code search"](https://www.youtube.com/watch?v=CqZA_KmygKw)

**Reply by traversaro** [reply]
Thanks @LiteBrite82 ! I may be wrong, but I could not find any reference to `third_party` folders being excluded from search results in those links. In the first link it is mentioned that "Vendored and generated code is excluded", but they do not specify how "vendored" code is identified. I guess `third_party` folder are classified as "vendored", but it would be great to have a clear documentation on what is considerd "vendored" code and what not.

**Reply by LiteBrite82** [reply]
I fully agree, Im still searching.

**Reply by hjpower2** [reply]
Eeffoc, coffee spelled backwards

**Reply by usmonster** [reply]
Possibly related & possibly more serious: #130254.

**Comment by jlisee** [comment]
Thanks for the quick response, I read over that page but not closely enough.

Just so you can understand the use case our `third_party` folder has some external code, but also a lot of build and other configuration related code we wrote.  If I see a reference in the main part of the tree to `external_repo_foo` it's great to be able to search and find that `external_repo_foo` is defined in the file we wrote `third_repo/external_repo_foo/BUILD`.

Eventually our `third_party` folder we consistent only of files like this, so at least for us the ability to simply force include a few paths no matter what [Enry](https://github.com/go-enry/go-enry) or a successor system thinks would be great.

**Reply by ghost** [reply]
+1 to indexing all code. I had to do some deep digging to find some error messages today; if we index all the vendor code it'll save me a lot of time (!).

**Comment by WaterSwallow** [comment]
> In our private repository on cs.github.com I can browse to files in our the `third_party` folder in the root of our repository but searches do not return any content there. I can even visit a file and see a detected symbol then click "search for .... in this repo" and it finds it everywhere _except_ the file I was just in.
> 
> Is there some global filtering control that is taking effect here that can be adjusted?

It seems like you're experiencing some difficulties with the search function in your private repository on cs.github.com, particularly with files in the third_party folder. 

The issue you're encountering, where search doesn't return content from specific directories, could potentially be due to some exclusion rules set in the .gitignore or .github/gitattributes files. These files inform Git (and therefore GitHub) which file patterns it should ignore, which may impact search results.

However, it's unusual that, even when clicking "search for .... in this repo" for a specific symbol within a file, the file itself isn't included in the search results. This behavior suggests something else may be affecting the search capability. 

If no exclusion rules have been set for the third_party folder in either .gitignore or .github/gitattributes files, it might be a more complex issue that needs further exploration. 

For deeper insights into this, I recommend you to connect with GitHub support directly, providing them with all the details of your issue. They can guide you through the possible solutions or escalate the issue if it's a bug in the search function.
